From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Thu, 24 Aug 2017 03:50:11 +0200
Subject: [PATCH] states: maas: Batch apply linux state with `-b 4`

Applying heavy states like linux.network on multiple nodes in
parallel might lead to spurious timeouts on slower systems.

Compromise between speed and realiability by applying this state
in batches of 4 nodes at once.

While at it, increase global command_timeout to 15s on Salt master,
since default command timeout (10s) is sometimes too small for all
minions to return within the required timeslot. Increase it by 50%.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 mcp/config/states/maas                                                  | 2 +-
 mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/mcp/config/states/maas b/mcp/config/states/maas
index 12ef9ed..697f83c 100755
--- a/mcp/config/states/maas
+++ b/mcp/config/states/maas
@@ -107,4 +107,4 @@ wait_for 10 "salt -C '* and not cfg01* and not mas01*' ssh.set_auth_key ${SUDO_U

 wait_for 10 "salt -C '* and not cfg01* and not mas01*' saltutil.sync_all"
 wait_for 10 "salt -C '* and not cfg01* and not mas01*' state.apply salt"
-wait_for 10 "salt -C '* and not cfg01* and not mas01*' state.apply linux,ntp"
+wait_for 10 "salt -C '* and not cfg01* and not mas01*' state.apply linux,ntp -b 4"
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml
index 7d95ebc..46526da 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/config.yml
@@ -37,6 +37,7 @@ parameters:
   salt:
     master:
       accept_policy: open_mode
+      command_timeout: 15
   reclass:
     storage:
       data_source:
